README.md

```diff
@@ -1,80 +1,627 @@
-
-## Configuration
+</details>
 
-Group mappings are resolved on the NameNode and the following configuration should be added to the NameNode role:
+## Group mapper
 
-### envOverrides
+Despites having the OPA authorizer described above there are a few use-cases to have a group mapper as well.
 
-#### HADOOP_CLASSPATH
+1. Correctly showing group information in HDFS, e.g. for file ownership.
+2. Only use the group mapper without the OAP authorizer
 
-* Fixed value of `"/stackable/hadoop/share/hadoop/tools/lib/*.jar"`
+Hadoop offers a few default group providers, such as:
 
-### configOverrides / `core-site.xml`
+* LDAP
+* Linux user group (usually provided by SSSD, Centrify or similar tools)
 
-#### hadoop.security.group.mapping
+Hadoop exposes an [interface](https://github.com/apache/hadoop/blob/rel/release-3.3.6/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/GroupMappingServiceProvider.java) that users can implement to extend these [group mappings](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/GroupsMapping.html). The Stackable Group Mapper does this to look up user groups from OPA.
 
-* Fixed value of `"tech.stackable.hadoop.StackableGroupMapper"`
+### Configuration
 
-#### hadoop.security.group.mapping.opa.url
+- Set `hadoop.security.group.mapping` in `core-site.xml` to `tech.stackable.hadoop.StackableGroupMapper`
+- Set `hadoop.security.group.mapping.opa.policy.url` in `core-site.xml` to the HTTP endpoint of your OPA rego rule, e.g. `http://opa.default.svc.cluster.local:8081/v1/data/hdfs/groups`
+- Make sure to not have set `hadoop.user.group.static.mapping.overrides` in `core-site.xml`, as this clashes with the information the group mapper provides.
 
-* The Opa Server endpoint e.g. `"http://test-opa.default.svc.cluster.local:8081/v1/data/hdfsgroups"`
+### API
 
-#### hadoop.security.group.mapping.opa.list.name
+The group mapper sends the following query to OPA:
 
-* Opa responses have a [root field](https://www.openpolicyagent.org/docs/latest/rest-api/#response-message) called `result`: the result itself - in this case of a list of user groups - is a top-level field within the root field and is configurable i.e. the group mapper just needs to look up this field from the response and this is passed in the configuration. This means that both the output format of the rego rule and the corresponding response field are configured independently of the group mapper.
+```json
+{
+  "input": {
+    "username": "alice"
+  }
+}
+```
 
-#### hadoop.user.group.static.mapping.overrides
+OPA needs to respond with the list of groups as follows:
 
-* The hdfs-operator will add a default static mapping whenever kerberos is activated. This should be removed so that the mapping implementation can provide this information instead: i.e. with an empty string `""`
+```json
+{
+  "result": {
+    "groups": {
+      "groups": [
```

Comment:
I don't think this is correct as the mapper now expects a list and not a map.

Reply:
c1aaaf6